boards/shields/seeed_xiao_expansion_board/doc/index.rst

```diff
@@ -0,0 +1,84 @@
+.. _seeed_xiao_expansion_board:
+
+Seed Studio XIAO Expansion Board
```

Comment:
Needs s/Seed/Seeed/ in the entire changeset

Reply:
Done.